pkg/registry/batch/job/strategy_test.go

```diff
@@ -496,7 +514,7 @@ func TestJobStrategy_PrepareForCreate(t *testing.T) {
 				ObjectMeta: getValidObjectMeta(1),
 				Spec: batch.JobSpec{
 					Selector:             validSelector,
-					Template:             validPodTemplateSpec,
+					Template:             expectedPodTemplateSpec,
```

Comment:
I think you should keep this the same to minimize the diff.  No reason to change this.  

I think it makes sense to call it valid as you aren't really testing these values.

Reply:
Actually, I want to change the podSpec in all of the wanted jobs, as I want to assert that each one contains the added labels.

I can rename it to `validXXX` but my point was that I create another `podTemplate` which contains the expected labels, hence why I named it `expectedPodTemplateSpec`

Reply:
I would minimize the changes honestly.  In this case, we have the `Selector` is set to `validSelector` but `Template` has `expectedPodTemplateSpec`.  We have the code as `wantJob` for expected so I don't think you need to change it.

Reply:
It will work just the same if we use `validPodTemplateSpec`, but then the test works because of a side-effect that we pass the `validPodTemplateSpec` as reference and the `generateSelectors` function adds the missing fields.
IMHO it is more explicit this way and clearer what is expected.

Reply:
My issue is that you have some cases where the values are valid and others that are expected.

```
					Selector:         validSelector,
					Template:         expectedPodTemplateSpec,
```

Its a nonblocking nit.  Just a suggestion because now we have different names for different items in the expected spec.